packages/zoe/src/contractSupport/zoeHelpers.js

```diff
@@ -99,20 +101,18 @@ export const makeZoeHelpers = (zcf) => {
      *
      */
     canTradeWith: (leftOfferHandle, rightOfferHandle) => {
-      const { issuerKeywordRecord } = zcf.getInstanceRecord();
-      const keywords = getKeys(issuerKeywordRecord);
-      const amountMaths = zcf.getAmountMaths(keywords);
       const { proposal: left } = zcf.getOffer(leftOfferHandle);
       const { proposal: right } = zcf.getOffer(rightOfferHandle);
-      const satisfied = (want, give) =>
-        keywords.every(keyword => {
-          if (want[keyword]) {
-            return amountMaths[keyword].isGTE(give[keyword], want[keyword]);
-          }
-          return true;
+      const leftAllocation = zcf.getCurrentAllocation(leftOfferHandle);
+      const rightAllocation = zcf.getCurrentAllocation(rightOfferHandle);
+      const satisfied = (want, availableForTrade) =>
+        getKeywords(want).every(keyword => {
+          const amountMath = helpers.getAmountMath(keyword);
+          return amountMath.isGTE(availableForTrade[keyword], want[keyword]);
```

Comment:
I don't see any tests that show that this change made a difference in behavior. I'd expect to see a test where the return value from canTradeWith changes when the currentAllocation changes. It might even be good to have one where it changes from true to false, and another for which it changes from false to true.

Reply:
Sure, sounds good. I can add those.